src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -315,12 +348,34 @@ public override int Execute(DocumentsOperationContext context)
                                 Reply.Add(new DynamicJsonValue
                                 {
                                     ["Key"] = cmd.Key,
-                                    ["Method"] = "DELETE",
+                                    ["Type"] = CommandType.DELETE.ToString(),
                                     ["Deleted"] = deleteResults.Count > 0
                                 });
                             }
                             
                             break;
+                        case CommandType.AttachmentPUT:
+                            using (var attachmentStream = AttachmentStreams.Dequeue())
+                            {
+                                var attachmentPutResult = Database.DocumentsStorage.AttachmentsStorage.PutAttachment(context, cmd.Key, cmd.Name, cmd.ContentType, attachmentStream.Hash, cmd.Etag, attachmentStream.File);
+                                LastEtag = attachmentPutResult.Etag;
+
+                                // Make sure all the metadata fields are always been add
+                                var attachmentPutReply = new DynamicJsonValue
+                                {
+                                    ["Type"] = CommandType.AttachmentPUT.ToString(),
+                                    [Constants.Documents.Metadata.Id] = attachmentPutResult.DocumentId,
+                                    ["Name"] = attachmentPutResult.Name,
+                                    [Constants.Documents.Metadata.Etag] = attachmentPutResult.Etag,
+                                    ["Hash"] = attachmentPutResult.Hash,
+                                    ["ContentType"] = attachmentPutResult.ContentType,
```

Comment:
No, in the server side we convert null to empty string, as PutAttachment should take only empty string (It calls GetLowerKeySliceAndStorageKey on the contentType).